sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -859,7 +857,7 @@ protected override async Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpo
                 var eventArgs = new PartitionInitializingEventArgs(partition.PartitionId, DefaultStartingPosition, cancellationToken);
                 await _partitionInitializingAsync(eventArgs).ConfigureAwait(false);
 
-                PartitionStartingPositionDefaults[partition.PartitionId] = eventArgs.DefaultStartingPosition;
+                PartitionStartingPositionDefaults.AddOrUpdate(partition.PartitionId, eventArgs.DefaultStartingPosition, (key, pos) => eventArgs.DefaultStartingPosition);
```

Comment:
Since we aren't basing the updated value on the existing value, I think we're better off setting the value indiscriminately using the indexer like before.

Reply:
Fair point; reverted the change.